Adds large tests

jcooklin opened the pull request:
In this PR we 
* Adds large tests that test the example
 * Tests can be deployed through docker-compose or kubernetes 
* Updates travis to deploy artifacts 
* Adds readme to scripts and examples

`K8S_TEST=1 make test-large`
![k8s_test](https://camo.githubusercontent.com/cc299f6f73b7e8bd6b130bd2b222918c18b0473f/687474703a2f2f692e67697068792e636f6d2f336f7a38784e6a636a4361716570616e76792e676966)

`examples/run-mock-influxdb.sh`
![example](https://camo.githubusercontent.com/f443c7265e469d5dcf85eb2fea4f7184dc6ac23e/687474703a2f2f692e67697068792e636f6d2f6c32537138703757796732726c49324a322e676966)

cc: @marcin-krolik @nanliu @intelsdi-x/plugin-maintainers

marcin-krolik commented on examples/tasks/mock-influxdb.sh:
```diff
@@ -0,0 +1,42 @@
+set -e
+set -u
+set -o pipefail
+
```
What do you think about adding check for connection to InfluxDB before anything else? Currently when one runs this script it will fail with
```bash
curl: (6) Could not resolve host: influxdb
```

just after downloading binaries. I think it would be more expressive to a user to inform about InfluxDB connectivity.

marcin-krolik commented:
In general directory structure is little bit confusing. Right now there's a lot of mixed purpose scripts in `scripts`. Additionally there are yaml files in in both `scripts` and in `examples/tasks`. What do you think about separating scripts by its purpose, like "test" , "deploy", configuration like yamls in "configuration"? 

Something similar to:
```
- scripts
   |-- tests
       |-- large
       |-- medium
       |-- ...
  |-- deployment
  |-- configuration
```

nanliu commented on examples/tasks/docker-compose.yml:
```diff
@@ -0,0 +1,23 @@
+      - "8083"
+      - "8086"
+      - "8090"
+      - "8099"
```
What are port 8090 and 8099 for?

jcooklin replied:
They are required if influxdb is configured in a cluster which we aren't so I'll remove them.

nanliu commented on examples/tasks/.setup.sh:
```diff
@@ -0,0 +1,25 @@
+function finish {
+  cd $__dir && docker-compose down
+}
+trap finish EXIT
```
Maybe trap SIGINT as well.

nanliu commented on examples/tasks/.setup.sh:
```diff
@@ -0,0 +1,25 @@
+fi
+
+# start the influxdb container
+cd $__dir && docker-compose up -d 
```
If you don't want the cd to occur if the docker-compose command fails, you need to wrap the entire line with ( ).

```
~
❯ (cd /tmp && fail)
zsh: command not found: fail

~
❯ cd /tmp && fail
zsh: command not found: fail

/tmp
❯
```

nanliu commented on scripts/large.sh:
```diff
@@ -37,9 +30,17 @@ _docker_project () {
 _docker_project docker-compose build
 _debug "running docker compose images"
-_docker_project docker-compose up -d
+_docker_project docker-compose up -d 
```
extra space at end

nanliu commented on scripts/large_compose.sh:
```diff
@@ -0,0 +1,35 @@
+docker_folder="${__proj_dir}/scripts/docker/${TEST_TYPE}"
+
+_debug "running docker compose images"
+cd ${__proj_dir}/examples/tasks && docker-compose up -d 
```
wrap ( )

nanliu commented on scripts/large_compose.sh:
```diff
@@ -0,0 +1,35 @@
+set -e
+echo "exit code from large_jenkins $test_res"
+_debug "stopping and removing containers"
+cd ${__proj_dir}/examples/tasks && docker-compose down
```
same as above

nanliu commented on scripts/large_tests.sh:
```diff
@@ -0,0 +1,50 @@
+_info "running the example ${__proj_dir}/examples/tasks/mock-influxb.sh"
+export PLUGIN_PATH="/etc/snap/path"
+export BIN_PATH="/usr/local/bin"
+source ${__proj_dir}/examples/tasks/mock-influxdb.sh
```
double quote file

nanliu commented on scripts/medium.sh:
```diff
@@ -0,0 +1,45 @@
+[[ -d ${influx_folder} ]] || _error "invalid Influxdb version: ${INFLUXDB_VERSION}"
+
+_docker_project () {
+  cd "${docker_folder}" && "$@"
```
wrap ( )

nanliu commented on scripts/test.sh:
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash 
```
extra space at end

nanliu commented on scripts/influxdb-deployment.yml:
```diff
@@ -0,0 +1,46 @@
+  name: influxdb
+spec:
+  ports:
+  - name: port-1  
```
might be more descriptive to label this port 'influx_http'  and 8083 'influx_admin'

nanliu commented on scripts/large.sh:
```diff
@@ -1,32 +1,25 @@
-#!/bin/bash
+#!/bin/bash 
```
extra space at the end

nanliu commented:
mostly lint related comments, otherwise LGTM.

jcooklin commented:
@nanliu before I merge will you review again?  I switched to using the image `intelsdi/snap:latest_alpine` and addressed your comments.

Also, please checkout https://github.com/intelsdi-x/snap-docker/pull/3.  It needs to land before this should be merged.

nanliu commented:
LGTM
